pkg/agent/controller/traceflow/packetin.go

```diff
@@ -0,0 +1,207 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package traceflow
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/contiv/libOpenflow/openflow13"
+	"github.com/contiv/ofnet/ofctrl"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/klog"
+
+	opsv1alpha1 "github.com/vmware-tanzu/antrea/pkg/apis/ops/v1alpha1"
+)
+
+type JSONPatch struct {
+	Op    string      `json:"op"`
+	Path  string      `json:"path"`
+	Value interface{} `json:"value"`
+}
+
+func (c *Controller) ReceivePacketIn(stopCh <-chan struct{}) {
+	ch := make(chan *ofctrl.PacketIn)
+	c.ofClient.SubscribePacketIn(1, ch)
```

Comment:
Use a constant for the "reason" field?

Reply:
Added OFPR_ACTION ofp_packet_in_reason = 1 /* Action explicitly output to controller. */